lib/load_brocfile.js

```diff
@@ -29,7 +30,12 @@ module.exports = function loadBrocfile(options) {
   process.chdir(baseDir);
 
   // Load brocfile
-  const brocfile = require(brocfilePath);
+  let brocfile = esmRequire(brocfilePath);
+
+  // Esm imports require an "export default" that is returned as { default: ... }
+  if (typeof brocfile === 'object' && brocfile.default) {
```

Comment:
@jdalton I noticed that

```js
export default “foo”;
```
when run should esm returns
```js
{
    default: “foo”
}
```
Does this make sense then?

Reply:
Yep. Remember `null` is typeof `"object"` too. There is an undocumented option tentatively called `dedefault` which will return `default` export in cases where you only export a `default` binding.
```js
const esmRequire = require('esm')(module, {
  cjs: {
    dedefault: true
  }
});
```

Reply:
Though if you're wanting to support `.mjs` too (`esm` options don't extend to `.mjs`) you might keep your existing style check. Your check could be refined to:
```js
if (typeof brocfile === 'object' && brocfile !== null) {
  const names = Object.keys(brocfile)

  if (names.length === 0 && names[0] === 'default') {
    brocfile = brocfile.default;
  }
}
```

Reply:
Yeah that's an option, I'd like to keep forwards compatibility with .mjs.
I'm not sure what should happen with the following:
```js
export const FOO = 'foo';
export default tree;
```
which will produce
```js
{
  "FOO": "foo",
  "default": tree
}
```
We can either error if there are additional keys or ignore them. Technically exporting a named export won't do anything as the importing side is only looking at default so we might want to just ignore them, in which case your suggested change above would miss.

Reply:
@oligriffiths - I'd prefer to ignore them. It seems perfectly reasonable to me for folks to have other shared exports from `Brocfile.js`...

Reply:
cool, they are ignored by default